### PR TITLE
Refactoring scheduled event to store instant instead of zoned tme zone

### LINKED ISSUE #38620: [ML] ScheduledEvent takes and stores ZonedDateTime but then discards the timezone
(Noticed while fixing #38506)

The `ScheduledEvent` class's constructor takes `ZonedDateTime` for both `startTime` and `endTime`.  However, when serialising both on the wire and as `XContent` it ditches the time zone and converts to an `Instant`.  It would make more sense to store `Instant`, take `Instant`s in the constructor and return `Instant`s from the accessors.  Then it would be completely clear to all users that the class was not preserving the time zone.

### PR DESCRIPTION
**Pull request by Mehrank64**
The ScheduledEvent class has never preserved the time
zone so it makes more sense for it to store the start and
end time using Instant rather than ZonedDateTime.

Closes #38620

### CODE REVIEW

**Comment by benwtrent** [comment]
Jenkins retest this please

**Comment by benwtrent** [comment]
@Mehrank64 Thank you so much for opening the PR 🚀 . Here are some gradle tasks to see if all the preCommit linting passes and the basic unit tests.
```
./gradlew :x-pack:plugin:core:preCommit :x-pack:plugin:ml:preCommit :x-pack:plugin:core:test :x-pack:plugin:ml:test
```
It looks like some linting failures are causing the builds to fail.

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/calendars/ScheduledEventTests.java [hunk 1]
```diff
@@ -27,7 +26,7 @@
 public class ScheduledEventTests extends AbstractSerializingTestCase<ScheduledEvent> {
 
     public static ScheduledEvent createScheduledEvent(String calendarId) {
-        ZonedDateTime start = DateUtils.nowWithMillisResolution();
+        Instant start = Instant.ofEpochSecond(Instant.now().toEpochMilli());
```
Comment (droberts195) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/calendars/ScheduledEventTests.java]: There's a discrepancy here - it's interpreting a millisecond number as seconds, resulting in a time long in the future.

**Comment by droberts195** [comment]
Regarding the wildcard imports, which the style checker doesn't like, there is advice on how to stop IntelliJ switching to wildcards in https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md:

> * IntelliJ: `Preferences/Settings->Editor->Code Style->Java->Imports`. There are two configuration options: `Class count to use import with '*'` and `Names count to use static import with '*'`. Set their values to 99999 or some other absurdly high value.

Hopefully if you change that setting and then re-save the files from IntelliJ it will remove the wildcard imports.

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/calendars/ScheduledEventTests.java (hunk 1 above)
Comment (Mehrank64) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/calendars/ScheduledEventTests.java]: It was definitely a mistake. just solved!

**Comment by Mehrank64** [comment]
> @Mehrank64 Thank you so much for opening the PR 🚀 . Here are some gradle tasks to see if all the preCommit linting passes and the basic unit tests.
> 
> ```
> ./gradlew :x-pack:plugin:core:preCommit :x-pack:plugin:ml:preCommit :x-pack:plugin:core:test :x-pack:plugin:ml:test
> ```
> It looks like some linting failures are causing the builds to fail.

**Comment by Mehrank64** [comment]
@benwtrent  I did resolve the linting failures and the now all the tests should pass. I closed the PR by accident and can't find any way to re-open it. will raise another PR.

**Comment by droberts195** [comment]
Jenkins retest this please

**Comment by benwtrent** [comment]
Jenkins retest this please

**Review by droberts195 — APPROVED** [review]
LGTM

Thanks for the contribution @Mehrank64 - much appreciated!

**Pull request merged**